tpm2/src/util/file_io.c

```diff
@@ -118,28 +138,69 @@ int read_bytes_from_file(char *input_path, uint8_t ** data, size_t *data_length)
   if (!BIO_read_filename(bio, input_path))
   {
     kmyth_log(LOG_ERR, "error opening input file: %s ... exiting", input_path);
-    BIO_free(bio);
+    if (!BIO_free(bio))
+    {
+      kmyth_log(LOG_ERR, "error freeing BIO ... exiting");
+    }
     return 1;
   }
 
   // Determine size of file
   struct stat st;
 
-  stat(input_path, &st);
+  if (stat(input_path, &st) == -1)
+  {
+    kmyth_log(LOG_ERR,
+              "input file (%s) stats could not be retrieved ... exiting",
+              input_path);
+    if (!BIO_free(bio))
+    {
+      kmyth_log(LOG_ERR, "error freeing BIO ... exiting");
+    }
+    return 1;
+  }
   int input_size = st.st_size;
 
+  // TODO: What maximum file size should be supported here?
```

Comment:
I don't think there was ever an intended max file size. If the file is too large to fit in memory, the malloc should fail.